pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -47,38 +70,54 @@ let
         ${replaced} = v;
         "tree-sitter-${replaced}" = v;
       }
-    ) generatedDerivations;
+    ) parsersWithMeta;
 
-  allGrammars = lib.attrValues generatedDerivations;
+  allGrammars = builtins.attrValues parsersWithMeta;
 
   # Usage:
   # pkgs.vimPlugins.nvim-treesitter.withPlugins (p: [ p.c p.java ... ])
   # or for all grammars:
   # pkgs.vimPlugins.nvim-treesitter.withAllGrammars
   withPlugins =
     f:
+    let
+      selectedGrammars = f (tree-sitter.builtGrammars // builtGrammars);
+
+      grammarPlugins = map grammarToPlugin selectedGrammars;
+
+      queryPlugins = lib.pipe selectedGrammars [
+        (map (g: g.passthru.associatedQuery or null))
+        (builtins.filter (q: q != null))
+      ];
```

Comment:
`withPlugins` is not the only way to install parsers. Which means, installing parsers using `nvim-treesitter-parsers` does not install queries, and it breaks.

Why should we even separate those in different derivations, just so we end up symlinking them? I know they are separate entities, but they are too tightly coupled together.

Reply:
> Also, a plugin may depend on a parser, but not nvim-treesitter, so only installing nvim-treesitter-parsers.xyz makes sense.

Plugins that depend on tree-sitter parsers typically don't depend on the queries provided by nvim-treesitter. They provide [their own queries](https://github.com/andymass/vim-matchup/blob/1c276e12b49a83c1bdca74351187b8adea5da4b9/after/queries).

kulala.nvim was the only exception I have come across, and that's not a hard dependency.
I personally think it would be better for plugins to be explicit about depending on nvim-treesitter queries.

Reply:
> Plugins that depend on tree-sitter parsers typically don't depend on the queries provided by nvim-treesitter. They provide their own queries.

I've checked every plugin that uses `nvim-treesitter-parsers`

- `haskell-scope-highlighting-nvim` provides `scope_highlighting.scm`
- `hurl-nvim` does not provide any queries
- `markdoc-nvim` does not provide any queries
- `neotest-gtest` does not provide any queries
- `nvim-nu` provides some queries
- `nvim-test` does not provide any queries

As you can see, most of them do not provide queries. Also, I think you said a parser without queries doesn't really make sense?